substrafl/algorithms/pytorch/torch_base_algo.py

```diff
@@ -152,14 +156,17 @@ def _local_predict(self, predict_dataset: torch.utils.data.Dataset, predictions_
 
         self._model.eval()
 
-        predictions = torch.Tensor([]).to(self._device)
-        with torch.inference_mode():
+        predictions = []
+        with torch.no_grad():
             for x in predict_loader:
                 x = x.to(self._device)
-                predictions = torch.cat((predictions, self._model(x)), 0)
+                predictions.append(self._model(x))
+        predictions = torch.cat(predictions, 0)  # only 1 cat operation, fewer copies
```

Comment:
Good catch 👍 
If you want to move this to a separate PR, it can get merged before the next release.

Reply:
OK thanks I'll make a separate PR then

Reply:
Thanks 🙏